macOSUpgrade.sh

```diff
@@ -230,17 +230,23 @@ validate_power_status() {
 }
 
 validate_free_space() {
-    local installerMajor diskInfoPlist freeSpace requiredDiskSpaceSizeGB
+    local installerMajor diskInfoPlist freeSpace requiredDiskSpaceSizeGB installerPath installerSizeBytes
 
     installerMajor="$1"
+    installerPath="$2"
 
     diskInfoPlist=$(/usr/sbin/diskutil info -plist /)
-
     ## 10.13.4 or later, diskutil info command output changes key from 'AvailableSpace' to 'Free Space' about disk space.
     freeSpace=$(
     /usr/libexec/PlistBuddy -c "Print :FreeSpace" /dev/stdin <<< "$diskInfoPlist" 2>/dev/null || /usr/libexec/PlistBuddy -c "Print :AvailableSpace" /dev/stdin <<< "$diskInfoPlist" 2>/dev/null
     )
 
+    ## The free space calculation also includes the installer, so it is excluded.
+    if [ -e "$installerPath" ]; then
+        installerSizeBytes=$(/usr/bin/du -s "$installerPath" | /usr/bin/awk '{print $1}' | /usr/bin/xargs)
+        freeSpace=$((freeSpace + installerSizeBytes))
```

Comment:
On my understanding, `freeSpace` must be calculated under startup disk with OS installer.app.
Event if `freeSpace` looks enough, there is no OS installer.app, it may be got smaller after download the installer.

So if  the installer.app exists, we don't take care about it here.
And if there is not the installer.app, could not calculate its size here.

Your point of this PR is very important. But your code looks different from what you wanted to do actually.

Reply:
No, I'm wrong. Sorry.
Now under reviewing.